### PR TITLE
Add a feature flag in UserAgent for AWS backend to monitor the adoptions

### PR DESCRIPTION
**Pull request by mxiamxia**
**Description:** <Describe what has changed.>
Add a feature flag in UserAgent for AWS backend to monitor the adoptions

### CODE REVIEW

File: exporter/awsemfexporter/cwlog_client.go [hunk 1]
```diff
@@ -175,9 +176,13 @@ func (client *cloudWatchLogClient) CreateStream(logGroup, streamName *string) (t
 	return "", nil
 }
 
-func newCollectorUserAgentHandler(buildInfo component.BuildInfo) request.NamedHandler {
+func newCollectorUserAgentHandler(buildInfo component.BuildInfo, logGroupName string) request.NamedHandler {
+	fn := request.MakeAddToUserAgentHandler(collectorDistribution, buildInfo.Version)
+	if strings.Contains(logGroupName, "ContainerInsights") {
```
Comment (pxaws) [inline exporter/awsemfexporter/cwlog_client.go:181]: Should we be more specific about the pattern of the log group name and maybe check the metric namespace as well? See https://github.com/aws-observability/aws-otel-collector/blob/main/deployment-template/eks/otel-container-insights-infra.yaml#L80-L81 and https://github.com/aws-observability/aws-otel-collector/blob/main/config/eks/prometheus/config-all.yaml#L229-L230 and https://github.com/aws-observability/aws-otel-collector/blob/main/config/ecs/container-insights/otel-task-metrics-config.yaml#L181-L186

Comment (mxiamxia) [reply exporter/awsemfexporter/cwlog_client.go]: added the regex to matches all ContainerInsight log group patterns below
```
/aws/containerinsights/<cluster-name>/prometheus
/aws/containerinsights/<eks-cluster-name>/performance
/aws/ecs/containerinsights/<ecs-cluster-name>/performance
```

File: exporter/awsemfexporter/cwlog_client.go [hunk 2]
```diff
@@ -175,9 +176,19 @@ func (client *cloudWatchLogClient) CreateStream(logGroup, streamName *string) (t
...
 	return request.NamedHandler{
 		Name: "otel.collector.UserAgentHandler",
-		Fn:   request.MakeAddToUserAgentHandler(collectorDistribution, buildInfo.Version),
+		Fn:   fn,
 	}
 }
+
+func matchContainerInsightsPattern(logGroupName string) bool {
+	regexP := "^\\/aws\\/.*containerinsights\\/.*\\/(performance|prometheus)$"
+	r, _ := regexp.Compile(regexP)
```
Comment (pxaws) [inline exporter/awsemfexporter/cwlog_client.go:192]: The compiled regex can be saved so that we don't need to compile it for each request.

Comment (mxiamxia) [reply exporter/awsemfexporter/cwlog_client.go:192]: it's a good callout but this check will be only execute once per EMF Exporter instance. I think it is fine in this case. :)

**Review by pxaws — APPROVED** [review]
LGTM

File: exporter/awsemfexporter/cwlog_client.go [hunk 3]
```diff
@@ -175,9 +176,19 @@ func (client *cloudWatchLogClient) CreateStream(logGroup, streamName *string) (t
...
+	}
 	return request.NamedHandler{
 		Name: "otel.collector.UserAgentHandler",
-		Fn:   request.MakeAddToUserAgentHandler(collectorDistribution, buildInfo.Version),
+		Fn:   fn,
 	}
 }
+
+func matchContainerInsightsPattern(logGroupName string) bool {
+	regexP := "^\\/aws\\/.*containerinsights\\/.*\\/(performance|prometheus)$"
```
Comment (straussb) [inline exporter/awsemfexporter/cwlog_client.go:191]: Just curious, why do you need to escape the forward slashes?  Wouldn't `"^/aws/.*containerinsights/.*/(performance|prometheus)$"` work?

**Review by straussb — APPROVED** [review]

File: exporter/awsemfexporter/cwlog_client.go (hunk 3 above)
Comment (mxiamxia) [reply exporter/awsemfexporter/cwlog_client.go]: You're right, it was only required when I tested the regex in some web regex editor.
Made the update, thx.

**Review by straussb — APPROVED** [review]

**Review by anuraaga — APPROVED** [review]

**Comment by mxiamxia** [comment]
Hi @bogdandrutu , please help to review and merge the PR. Thanks!

**Pull request merged**